inc/wp-custom-menu.php

```diff
@@ -8,6 +8,355 @@
+					$column[ $m->ID ]['href']        = $m->url;
+					$column[ $m->ID ]['has_current'] = false;
+
+					/**
+					 * Add this item's data as a child to the $dropdown array we created in step 2.
+					 * Place it under the parent, then under 'items', in a new array with ID of this item's ID.
+					 */
+					$dropdown[ $m->menu_item_parent ]['items'][ $m->ID ] = $column[ $m->ID ];
+					
+					/**
+					 * Determine this item's top-menu item (grandparent) by getting the parent ID of this item's parent.
+					 * Adding a check here to ensure that there is a parent array in the parent of this item for us to
+					 * add anything to.
+					 */
+					if ( array_key_exists( 'parent', $dropdown[ $m->menu_item_parent ] ) ) {
+						$top_menu = $dropdown[ $m->menu_item_parent ]['parent'];
+			 /* At step 3, if menu items are still leftover, we have a column header above our menu items
+			  *	column headers are not processed in previous steps, they are available items in $pre_menu, but filtered out
+			  *	of $menu in step 2. column headers are children of the top menu, menu items are children of column headers 
+			  */
+						$middle_menu = $dropdown[ $m->menu_item_parent ];
+
+			 if( ! in_array([ $middle_menu->ID ]['ID'], $our_array_menu) ) {
+				$middle_menu_column = array();
+				$middle_menu_column[ $middle_menu->ID ]                = array();
+			   $middle_menu_column[ $middle_menu->ID ]['ID']          = $middle_menu['ID'];
+			   $middle_menu_column[ $middle_menu->ID ]['type']        = "heading";
+			   $middle_menu_column[ $middle_menu->ID ]['text']        = $middle_menu['text'];
+			   //$middle_menu_column[ $middle_menu->ID ]['href']        = "#";
+			   $middle_menu_column[ $middle_menu->ID ]['has_current'] = false;
+
+			   $pre_menu[ $top_menu ]['items'][ $m->menu_item_parent ]['items'][ $middle_menu->ID ] = $middle_menu_column[ $middle_menu->ID ];
+
+			 }
+					
+						// The menu link can be relative or absolute.
+						// Format menu link and remove absolute base url from link
+						$prefix = get_home_url();
+						$menu_url = rtrim( $m->url, '/' ) . '/';
+						if ( 0 === strpos( $menu_url, $prefix ) ) {
+								$menu_url = substr( $menu_url, strlen( $prefix ) );
+						}
+						$menu_url = $subsite_base_folder . $menu_url;
+
+						if ( $current_uri === $menu_url ) {
+							$pre_menu[ $top_menu ]['has_current'] = true;
+						}
+
+						$pre_menu[ $top_menu ]['items'][ $m->menu_item_parent ]['items'][ $m->ID ] = $column[ $m->ID ];
+					}
+				}
+			}
+			// The UDS nav menu requires that we re-format our menu.
+			// We must reset the menu IDs from the array keys to sequential array keys, 0 to x.
+			// And the items[] nested arrays must be wrapped in an additional array.
+			$menu['nav-items'] = array();
+			$menu['cta-buttons'] = $cta_buttons;
+			$menu['logo-override'] = $asu_logo_override_array;
+			$menu['show-partner-logo'] = $show_partner_logo;
+			$menu['partner-logo'] = $add_partner_logo_array;
```

Comment:
These two lines trigger PHP warnings for undefined variables. I think they probably need to be initialized as empty arrays in line 40 and 49 before things are assigned to them? (Or, set an empty array as the `else` part of that logic?)

<img width="1508" alt="image" src="https://user-images.githubusercontent.com/2085753/204699792-0b715a3d-ad67-44c0-b3c0-f65dd23cb618.png">